Fix handling of long running connections (from Incus)

markylaing opened the pull request:
When connecting to LXD via a load-balancer, long-polling LXD often fails because the connection is timing out. This PR mitigates the issue by adding a keep-alive timeout on the LXD server.

markylaing commented:
The CLA checker failed a few times because I used the British spelling `Licence` instead of `License` :facepalm:

tomponline commented:
@markylaing would you mind adding some additional info to the PR description to describe what "fix" means in this context? Thanks

markylaing commented:
> @markylaing would you mind adding some additional info to the PR description to describe what "fix" means in this context? Thanks

Yep of course, apologies

markylaing commented:
@tomponline I've updated the description. I've also removed the second commit related to OIDC. I'm not aware of this causing any issues and it's an easy fix if it is ever reported.

tomponline commented:
> When connecting to LXD via a load-balancer, long-polling LXD often fails because the connection is timing out. This PR mitigates the issue by adding a keep-alive timeout on the LXD server.

Do you know what the default IdleTimeout value was?

markylaing commented:
> > When connecting to LXD via a load-balancer, long-polling LXD often fails because the connection is timing out. This PR mitigates the issue by adding a keep-alive timeout on the LXD server.
> 
> Do you know what the default IdleTimeout value was?

The go doc has:
```
// IdleTimeout is the maximum amount of time to wait for the
// next request when keep-alives are enabled. If IdleTimeout
// is zero, the value of ReadTimeout is used. If both are
// zero, there is no timeout.
```

The ReadTimeout is not set so it's currently defaulting to zero.

tomponline commented:
> The ReadTimeout is not set so it's currently defaulting to zero.

So if not timeout now, then how come "long-polling LXD often fails because the connection is timing out. "?

markylaing commented:
I'm going to close this because I can't see any issues relating to what it purports to fix.
